File: server/src/main/java/com/yoedu/job_board_platform/models/Company.java
package com.yoedu.job_board_platform.models;

import java.time.OffsetDateTime;
import java.util.UUID;

import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EntityListeners;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "companies")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@EntityListeners(AuditingEntityListener.class)
public class Company {
    @Id
    @Column(columnDefinition = "uuid")
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "company_name", nullable = false, unique = true, length = 100)
    private String companyName;

    @Column(nullable = false, unique = true)
    private String slug;

    @Column(nullable = false, columnDefinition = "text")
    private String address;

    @Column(columnDefinition = "text")
    private String description;

    @Column(length = 255)
    private String website;

    @Column(name = "logo_url", columnDefinition = "text")
    private String logoUrl;

    @Column(nullable = false, unique = true, length = 255)
    private String email;

    @Column(nullable = false, length = 15)
    private String phone;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    @Builder.Default
    private CompanyStatus status = CompanyStatus.PENDING;

    @Column(name = "tax_code", length = 20)
    private String taxCode;

    @Column(name = "rejection_reason", columnDefinition = "text")
    private String rejectionReason;

    @Column(name = "is_approved", nullable = false)
    @Builder.Default
    private boolean isApproved = false;

    @Column(name = "created_at", nullable = false)
    @CreatedDate
    private OffsetDateTime createdAt;

    @Column(name = "approved_at")
    private OffsetDateTime approvedAt;
}

File: client/src/types/company.ts
export interface CompanyRequest {
	companyName?: string;
	address?: string;
	description?: string;
	website?: string;
	logoUrl?: string;
	email?: string;
	phone?: string;
	taxCode?: string;
}

export interface CompanyResponse {
	id: string;
	companyName: string;
	slug: string;
	address: string;
	description: string | null;
	website: string | null;
	logoUrl: string | null;
	email: string | null;
	phone: string | null;
	taxCode: string | null;
	status: CompanyStatus;
	isApproved: boolean;
	createdAt: string;
	approvedAt: string | null;
}

export interface AdminPendingCompanyResponse extends CompanyResponse {
	employerName: string | null;
	employerEmail: string | null;
	employerPhone: string | null;
	roleInCompany: string | null;
}

export const CompanyStatus = {
	PENDING: "PENDING",
	APPROVED: "APPROVED",
	REJECTED: "REJECTED",
	SUSPENDED: "SUSPENDED",
} as const;

export type CompanyStatus = (typeof CompanyStatus)[keyof typeof CompanyStatus];
